server/meshmodel/helper.go

```diff
@@ -0,0 +1,135 @@
+	regManager       *meshmodel.RegistryManager
+	componentChan    chan v1alpha1.ComponentDefinition
+	relationshipChan chan v1alpha1.RelationshipDefinition
+	doneSignal       chan bool
+	errorChan        chan error
+}
+
+func (ch *ComponentHelper) Init(hc *models.HandlerConfig, rm *meshmodel.RegistryManager) {
+	ch.handlerConfig = hc
+	ch.regManager = rm
+	ch.componentChan = make(chan v1alpha1.ComponentDefinition, 1)
+	ch.relationshipChan = make(chan v1alpha1.RelationshipDefinition, 1)
+	ch.doneSignal = make(chan bool)
+	ch.errorChan = make(chan error)
+}
+
+// seed the local meshmodel components
+func (ch ComponentHelper) SeedComponents() {
+	// Watch channels and register components and relationships with the registry manager
+	go ch.watchComponents()
+
+	// Read component and relationship definitions from files and send them to respective channels
+	ch.generateComponents("/components")
+	ch.generateRelationships("/relationships")
+	ch.doneSignal <- true
+}
+
+func (ch ComponentHelper) generateComponents(pathToComponents string) {
+	path, err := filepath.Abs(pathToComponents)
+	if err != nil {
+		ch.errorChan <- errors.Wrapf(err, "error while getting absolute path for generating components")
+		return
+	}
+
+	err = filepath.Walk(path, func(path string, info fs.FileInfo, err error) error {
+		if info == nil {
+			return nil
+		}
+
+		if !info.IsDir() {
+			var comp v1alpha1.ComponentDefinition
+			byt, err := os.ReadFile(path)
+			if err != nil {
+				return err
+			}
+			err = json.Unmarshal(byt, &comp)
+			if err != nil {
+				return err
+			}
+			if comp.Metadata != nil && comp.Metadata["published"] == true {
+				utils.WriteSVGsOnFileSystem(&comp)
+				ch.componentChan <- comp
+			}
+		}
+		return nil
+	})
+	ch.errorChan <- errors.Wrapf(err, "error while generating components")
+	return
+}
+
+func (ch ComponentHelper) generateRelationships(pathToComponents string) {
+	path, err := filepath.Abs(pathToComponents)
+	if err != nil {
+		ch.errorChan <- errors.Wrapf(err, "error while getting absolute path for generating relationships")
+		return
+	}
+
+	err = filepath.Walk(path, func(path string, info fs.FileInfo, err error) error {
+		if info == nil {
+			return nil
+		}
+		if !info.IsDir() {
+			var rel v1alpha1.RelationshipDefinition
+			byt, err := os.ReadFile(path)
+			if err != nil {
+				return nil
+			}
+			err = json.Unmarshal(byt, &rel)
+			if err != nil {
+				return nil
+			}
+			ch.relationshipChan <- rel
+		}
+		return nil
+	})
+
+	ch.errorChan <- errors.Wrapf(err, "error while generating relationships")
+	return
+}
+
+func (ch ComponentHelper) watchComponents() {
+	var err error
+	for {
+		select {
+		case comp := <-ch.componentChan:
+			err = ch.regManager.RegisterEntity(meshmodel.Host{
+				Hostname: ArtifactHubComponentsHandler,
+			}, comp)
+		case rel := <-ch.relationshipChan:
+			err = ch.regManager.RegisterEntity(meshmodel.Host{
+				Hostname: ArtifactHubComponentsHandler,
+			}, rel)
+		case mhErr := <-ch.errorChan:
```

Comment:
@Azanul nil check?

Reply:
nil checks are already there before pushing err to the channel

Reply:
Nope. They are not. https://github.com/meshery/meshery/pull/7314/files#diff-ab4499e6d5cf39814e6b6dd3607e23566fe60b939e1bb2fbb068a832003046c4R77

Reply:
And even if they were. The consumer should always be responsible for nil checks to avoid panic

Reply:
> The consumer should always be responsible for nil checks to avoid panic

I'd agree in case of fetching data from an external source. Why would we follow this in case of internal system?

Reply:
nil checking multiple places vs nil checking once. We have multiple producers and one consumer.